File: src/main/java/com/lanka_guide/tripplanner/model/activity/transit/Transit.java
package com.lanka_guide.tripplanner.model.activity.transit;

import com.lanka_guide.tripplanner.model.activity.Activity;
import com.lanka_guide.tripplanner.model.place.transport.TransportNode;

import java.time.ZonedDateTime;

public class Transit extends Activity {
    private TransportNode transportNode;

    public Transit(ZonedDateTime from, ZonedDateTime to) {
        super(from, to);
    }
}

File: src/main/kotlin/com/lankaguide/tripplanner/model/activity/transit/Transit.kt
package com.lankaguide.tripplanner.model.activity.transit

import com.lankaguide.tripplanner.model.activity.Activity
import com.lankaguide.tripplanner.model.place.transport.TransportNode

import java.time.ZonedDateTime

open class Transit(from: ZonedDateTime, to: ZonedDateTime) : Activity(from, to) {
    private val transportNode: TransportNode? = null
}
